kolibri/auth/filters.py

```diff
@@ -0,0 +1,185 @@
+from django.db import models
+from django.db.models.query import F
+from six import string_types
+
+from .constants import collection_kinds
+
+
+class HierarchyRelationsFilter(object):
```

Comment:
> So, I think we'll want to (occasionally) run the full test suite using these alternative database engines, once we start using them, but don't think it makes sense to try to integrate them into the test suite itself (would probably make for much slower tests).

I agree that it could make for slower tests -- how *much* slower probably depends on a lot of factors. For instance, we may only wish to run a judiciously chosen *subset* of the full test suite with postgres. Or we could always carefully identify bottlenecks using a postgres test db and optimize for them e.g. by preserving the test database in between runs (avoiding the need to rebuild it every time). In fact, Django has some facilities for this already.

Moreover for me the takeaway from the case you mention regarding KA Lite is to test early and often with the target stack -- if you plan to use postgres as a backend *eventually*, IMO it would save time ultimately to start testing with it *now*. Otherwise we risk quietly building up a lot of changes that we'll need to slog through when we *do* finally want to use it. Bear in mind that Django has a lot of scrutiny and provides an ORM that's meant to provide a consistent interface to any database backend, yet *still* has differences between backends. In contrast this project has orders of magnitude less scrutiny, and this code in particular is presumably being scrutinized only by you and me.

Personally, because I'm not an expert with postgres and I'm not able to clearly identify what will be problematic and what will not, I would hedge my bets by testing early, often, and automatically as opposed to later, occasionally, and manually. That's my 2 cents. If you don't find my reasoning compelling enough, I'll merge this.